drivers/platform/xilinx/xilinx_gpio_irq.c

```diff
@@ -93,7 +93,8 @@ static void xil_gpio_irq_handler(struct xil_gpio_irq_desc *param)
 			XGpioPs_IntrDisablePin(&param->my_Gpio, callback_desc->pin_nb);
 			XGpioPs_IntrClearPin(&param->my_Gpio, callback_desc->pin_nb);
 			callback_desc->callback.callback(callback_desc->callback.ctx, 0U, NULL);
-			XGpioPs_IntrEnablePin(&param->my_Gpio, callback_desc->pin_nb);
+			if(param->retriggerable)
```

Comment:
I think the change was not needed. You can disable the interrupt with irq_disable rigth?
Otherwise will be trigger for every event

Reply:
If you want to disable the interrupt within the irq_handler, using irq_disable won't work because, after the function return, XGpioPs_IntrEnablePin will be called (neutralizing irq_disable effect).

Reply:
You can us a flag set in irq_disable in check that.  For example in xil_callback_desc  have a bool: Is enabled.

Reply:
As you are saying, because you also tried :) is to disable the interrupt from the irq_handler so this is what the use will expect too.